eth_account/_utils/transaction_utils.py

```diff
@@ -0,0 +1,82 @@
+from typing import (
+    Any,
+    Dict,
+    Sequence,
+)
+
+from toolz import (
+    assoc,
+    dissoc,
+)
+
+from eth_account._utils.validation import (
+    is_rlp_structured_access_list,
+    is_rpc_structured_access_list,
+)
+
+
+def possibly_set_transaction_type(transaction_dict: Dict[str, Any]) -> Dict[str, Any]:
+    if 'type' not in transaction_dict:
+        if 'gasPrice' in transaction_dict and 'accessList' in transaction_dict:
+            # access list txn - type 1
+            transaction_dict = assoc(transaction_dict, 'type', '0x1')
+        elif 'maxFeePerGas' in transaction_dict and 'maxPriorityFeePerGas' in transaction_dict:
+            # dynamic fee txn - type 2
+            transaction_dict = assoc(transaction_dict, 'type', '0x2')
```

Comment:
If this stays here though, if there is no type it will be validated as a legacy transaction. The utility method only _adds_ the `type` to the dictionary if the type is implicit. I didn't expect to validate anything here but we could try to change some of this around if it makes more sense.

Reply:
Ahh, okay. Yeah, that makes sense to me! I forgot you didn't _have_ to pass in the type